packages/integration/lib/src/views/post/post_detail_screen.dart

```diff
@@ -168,372 +146,121 @@ class _LMFeedPostDetailScreenState extends State<LMFeedPostDetailScreen> {
-                body: Center(
-                  child: Text(
-                    '${snapshot.error}',
-                    style: Theme.of(context).textTheme.titleLarge,
-                  ),
-                ),
+        child: BlocListener(
+          listener: (context, state) {
+            if (state is LMFeedPostUpdateState) {
+              if (state.postId == widget.postId) {
+                if (state.actionType == LMFeedPostActionType.pollSubmit) {
+                  postData = LMFeedPostUtils.updatePostData(
+                    postViewData: state.post ?? postData!,
+                    actionType: state.actionType,
+                    commentId: state.commentId,
+                    pollOptions: state.pollOptions,
+                  ).copyWith();
+                } else {
+                  LMFeedPostUtils.updatePostData(
+                    postViewData: state.post ?? postData!,
+                    actionType: state.actionType,
+                    commentId: state.commentId,
+                    pollOptions: state.pollOptions,
+                  );
+                }
+                rebuildPostWidget.value = !rebuildPostWidget.value;
+              }
+            } else if (state is LMFeedEditPostUploadedState) {
+              LMPostViewData editedPost = state.postData.copyWith();
+              postData = editedPost;
+              rebuildPostWidget.value = !rebuildPostWidget.value;
+            } else if (state is LMFeedPostDeletionErrorState) {
+              LMFeedCore.showSnackBar(
+                context,
+                state.message,
+                _widgetSource,
               );
-            } else {
-              return const SizedBox.shrink();
             }
           },
+          bloc: postBloc,
+          child: ValueListenableBuilder(
+              valueListenable: rebuildPostWidget,
+              builder: (context, _, __) {
+                return _widgetBuilder.scaffold(
+                  source: _widgetSource,
+                  resizeToAvoidBottomInset: true,
+                  backgroundColor: feedTheme.backgroundColor,
+                  bottomNavigationBar: LMFeedBottomTextField(
+                    postId: widget.postId,
+                    focusNode: _commentFocusNode,
+                  ),
+                  appBar: widget.appBarBuilder?.call(context, defAppBar()) ??
+                      defAppBar(),
+                  body: Align(
+                    alignment: Alignment.topCenter,
+                    child: Container(
+                        width: screenWidth,
+                        child: CustomScrollView(
+                          slivers: [
+                            if (isDesktopWeb)
+                              SliverPadding(
+                                  padding: EdgeInsets.only(top: 20.0)),
+                            SliverToBoxAdapter(
+                              child: BlocBuilder<LMFeedCommentBloc,
```

Comment:
Is this the right BLoC? If so, why are we using this to build or rebuild the post widget?

Reply:
Here it is used to show comment count of post in appbar as i have mentioned in above comment.